react/src/components/molecules/HeaderSearch/index.js

```diff
@@ -19,16 +20,33 @@ class HeaderSearch extends React.Component {
   handleChange(event) {
     const query = event.target.value;
     this.setState({ value: query });
-    // invokes custom function if passed in the component
+    console.log(query)
+    /**
+       * Invokes a custom onChange function if passed.
+       * @param {string} query - The current query string of the input.
+    */
     if (typeof this.props.onChange === 'function') {
       this.props.onChange(query);
     }
   }
+  handleClick(e) {
+    if (is.fn(this.props.buttonSearch.onClick)) {
+      /**
+       * Invokes a custom onClick function if passed to the buttonSearch component.
+       * @param {object} event - The click event.
+       * @param {string} query - The current query string input.
+      */
+      this.props.buttonSearch.onClick({ event: e, query: this.state.value });
```

Comment:
You should not be passing the event as it will have expired for whatever set buttonSearch.onClick.